lib/std/build/RunStep.zig

```diff
@@ -136,6 +137,23 @@ pub fn expectStdErrEqual(self: *RunStep, bytes: []const u8) void {
     self.stderr_action = .{ .expect_exact = self.builder.dupe(bytes) };
 }
 
+/// Returns true if the step could be run, otherwise false
+pub fn isRunnable(
```

Comment:
I might have missed that, but why is this logic duplicated from [`cross_target.zig`](https://github.com/ziglang/zig/blob/1d55705fa42d83556346f8fcb3a8fce0d55f06ad/lib/std/zig/cross_target.zig#L608)?

Reply:
I'm not quite sure what exactly is duplicated here as CrossTarget declares the executors to use, and this code is just using those. In particular this helper function could be used in the RunStep to check if the step is viable (going to work) to avoid runtime errors.

Reply:
Oh sorry, what I meant to point out is that we already have the ability to run (cross-target) tests that require an emulator or similar, therefore I don't really see the need for this in `RunStep.zig`. See for instance output of `zig test --help`:

```
...
Test Options:
  --test-filter [text]           Skip tests that do not match filter
  --test-name-prefix [text]      Add prefix to all tests
  --test-cmd [arg]               Specify test execution command one arg at a time
  --test-cmd-bin                 Appends test binary path to test cmd args
  --test-evented-io              Runs the test in evented I/O mode
  --test-no-exec                 Compiles test binary without running it
...
```

We already have a way of specifying a custom execution command which is exactly how we invoke foreign tests on some host for instace, like Wasm in Wasmtime.

Reply:
Yes, but as I mentioned in the comment, the current RunStep behavior is inconsistent. I guess if it would be called NativeRunStep then it will be less confusing, but right now, then one specifies that it's possible to use QEMU, but RunStep gives IncorrectExe error, it's not immediately obvious that RunStep works with native target only.

Moreover, sometimes it's required to run the exe (not tests) in the emulator. Few use-cases for myself are:
 - complicated build processes (like LuaJIT compilation I've mentioned in the PR comment). `buildvm` (compiled from the source code using Zig in my toolchain) that is used for building LuaJIT VM doesn't support cross-compilation, and it's required to be invoked with the target architecture (i.e. to produce 32-bit LuaJIT, I have to use 32-bit `buildvm`). It can't be run as a test as it's standalone program and doesn't export anything to be linked into tests. Please take a look to [build.zig](https://gitlab.com/slpm/slpm/-/blob/main/modules/native/luajit/build.zig) for the reference.
 - cross-platform benchmarks. I'm working with the embedded systems, and need to measure the libraries footprint (e.g. pcre vs hyperscan for regexes). The benchmarks are compiled into executables for the deployment on the target platforms, and it's handy to run them first with QEMU to verify file sizes and memory consumption. While I could definitely use shell scripting for that, using extended RunStep functionality provided in this PR, makes the task much easier.

However, as I said, it's not critical as I have the workaround (like `ExecutorRunStep` in the referenced sources above). If you still have doubts, I can just remove it.

Reply:
Ah gotcha. Hmm, ok, I'll need to have @andrewrk take another look at this as we might already have a way to do this that I'm not aware of.